src/usr/local/buildpack/utils/ruby.sh

```diff
@@ -4,36 +4,38 @@ function gem_install() {
   tool_path=$(find_tool_path)
 
   if [[ -z "${tool_path}" ]]; then
-    INSTALL_DIR=$(get_install_dir)
-    base_path=${INSTALL_DIR}/${TOOL_NAME}
-    tool_path=${base_path}/${TOOL_VERSION}
+    tool_path=$(create_versioned_tool_path)
+    mkdir -p "${tool_path}"
 
-    mkdir -p ${tool_path}
-
-    GEM_HOME=$tool_path gem install  --bindir $tool_path/bin ${TOOL_NAME} -v ${TOOL_VERSION}
+    GEM_HOME=$tool_path gem install  --bindir "$tool_path"/bin "${TOOL_NAME}" -v "${TOOL_VERSION}"
 
   # TODO: clear gem cache
   fi
 }
 
 
 function gem_link_wrapper() {
-  link_wrapper ${1:-$TOOL_NAME} $tool_path/bin
+  link_wrapper "${1:-$TOOL_NAME}" "$tool_path/bin"
```

Comment:
Looks promising, but can we split out those changes as small fix Pr's, so we have less changes here?